core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1791,7 +1791,8 @@ private[spark] class DAGScheduler(
     // if the cluster manager explicitly tells us that the entire worker was lost, then
     // we know to unregister shuffle output.  (Note that "worker" specifically refers to the process
     // from a Standalone cluster, where the shuffle service lives in the Worker.)
-    val fileLost = workerLost || !env.blockManager.externalShuffleServiceEnabled
+    val fileLost = workerLost || (!env.blockManager.externalShuffleServiceEnabled &&
+        !sc.conf.get(config.EXTERNAL_SHUFFLE_ENABLED))
```

Comment:
With worker lost and EXTERNAL_SHUFFLE_ENABLED = true, fileLost should be false? Due to the shuffle files metadata doesn't reside in Worker.
I think this should be `!sc.conf.get(config.EXTERNAL_SHUFFLE_ENABLED) && (workerLost || (!env.blockManager.externalShuffleServiceEnabled))`

Reply:
Good catch--I had missed this since I never tested it in standalone mode (only YARN).